python/cudaq/kernel/kernel_decorator.py

```diff
@@ -343,9 +344,25 @@ def from_json(jStr, overrideDict=None):
 
     def __call__(self, *args):
         """
-        Invoke the CUDA-Q kernel. JIT compilation of the 
-        kernel AST to MLIR will occur here if it has not already occurred. 
+        Invoke the CUDA-Q kernel. JIT compilation of the kernel AST to MLIR 
+        will occur here if it has not already occurred, except when the target
+        requires custom handling.
         """
+
+        # Check if target is set
+        try:
+            target_name = cudaq_runtime.get_target().name
```

Comment:
I think since they are different scenarios they don't need to be combined in one block.
Practically, the way `LinkedLibraryHolder` works, I don't think the target will be `None`, this check is to ensure in case the function gets called otherwise.